plugins/module_utils/debug.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2019, XLAB Steampunk <steampunk@xlab.si>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import os
+import tempfile
+
+
+def log(message):
+    """
+    Log message to a file (/tmp/sensu-ansible.log) at remote target
+
+    Sensu API returns fairly modest error messages (e.g. when PUT payload contains
+    unsupported parameter, the error message won't tell you which one) and that
+    makes it difficult to debug. For that reason we decided to support at least
+    the most primitive type of logging: write to /tmp/sensu-ansible.log file.
+    Beware the log file resides on Ansible target and not host because this is
+    where the module gets executed.
+
+    This function won't do anything unless target has environment variable
+    SENSU_ANSIBLE_DEBUG set to "yes". When troubleshooting, just set the env
+    variable in the playbook.
+    """
+    if os.environ.get("SENSU_ANSIBLE_DEBUG", "").lower() in ["yes", "true"]:
```

Comment:
I agree that we should could the overhead, but it feels it would be nicer if we introduce module-level valiable `DEBUG` to only calculate the boolean once instead overloading the entire method, wdyt (I just pushed the implementation so you examine what I mean)? In both cases we now only access env once per task.

Reply:
This still executes `if` on each call, but since this only adds on average 4 `if` statement evaluations per Ansible task execution, I am fine with it. We can always optimize it later if we find this to be a bottleneck (which is highly unlikely).